portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,35 +26,63 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Comment;
 
-
-/** Servlet that returns some example content. */
+/** Servlet that returns data for my portfolio. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List comments = new ArrayList<String>();
+  private List comments = new ArrayList<Comment>();
+  private static final int DEFAULT_NUM_COMMENTS = 10;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Load from datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    int commentsMaxNum = DEFAULT_NUM_COMMENTS;
+    try {
+      commentsMaxNum = new Integer(request.getParameter("comments-num"));
+    } catch (NumberFormatException e) {
```

Comment:
You're totally right, it really returns a `NumberFormatException` for `null` input in the constructor. Thanks for checking!